liquibase-standard/src/main/java/liquibase/database/core/OracleDatabase.java

```diff
@@ -256,12 +256,12 @@ protected String getAutoIncrementClause(final String generationType, final Boole
 
     @Override
     public String generatePrimaryKeyName(String tableName) {
-        if (tableName.length() > 27) {
-            //noinspection HardCodedStringLiteral
-            return "PK_" + tableName.toUpperCase(Locale.US).substring(0, 27);
+        int maximumLength = getIdentifierMaximumLength();
```

Comment:
It's interesting that this method was added back in 2017 and never used.

Reply:
Yeap, probably the method was designed as part of some improvements and unfortunately forgot.